### PR TITLE
DT-1298 - Enhance JWT for legacy username attribute where required by client interacting with legacy system

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import org.springframework.security.oauth2.provider.token.TokenEnhancer
 import uk.gov.justice.digital.hmpps.oauth2server.security.AuthSource
 import uk.gov.justice.digital.hmpps.oauth2server.security.AuthSource.Companion.fromNullableString
 import uk.gov.justice.digital.hmpps.oauth2server.security.UserPersonDetails
-import java.util.Optional
+import java.util.*
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.kt:16]: not sure how this managed to pass `ktlint`?

Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.kt:16]: not sure how this managed to pass `ktlint`?  Maybe it doesn't...
You'll need to run
```
./gradlew ktlintApplyToIdea addKtlintFormatGitPreCommitHook
```
locally to set it up

Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/JWTTokenEnhancer.kt:16]: Installed ktlint after commit - sorry!

File: src/main/resources/templates/ui/form.html [hunk 2]
```diff
@@ -120,6 +120,18 @@ <h1 th:if="${param.client}" class="govuk-heading-l">Edit client '<span th:text="
       </div>
     </div>
 
+    <div class="govuk-form-group">
+      <div class="govuk-form-group">
+        <label class="govuk-label" for="databaseUsernameField">Legacy Username</label>
```
Comment (petergphillips) [inline src/main/resources/templates/ui/form.html:125]: Missed one :grin:

**Review by petergphillips — DISMISSED** [review]
LGTM

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**